app/src/main/java/com/asfoundation/wallet/di/ToolsModule.java

```diff
@@ -709,10 +703,21 @@ AsfInAppPurchaseInteractor provideAsfInAppPurchaseInteractor(
     return new PoASubmissionService(api);
   }
 
-  @Singleton @Provides BalanceGetter provideBalanceGetter(PoASubmissionService service) {
+  @Singleton @Provides BackendApi provideBackendApi(OkHttpClient client, Gson gson) {
+    return new Retrofit.Builder().baseUrl(BuildConfig.BACKEND_HOST)
+        .client(client)
+        .addConverterFactory(GsonConverterFactory.create(gson))
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+        .create(BackendApi.class);
+  }
+
+  @Singleton @Provides BalanceGetter provideBalanceGetter(BackendApi backendApi) {
```

Comment:
you should use AppcoinsRewards class instead of making a call directly to api. It has both methods you need:
-getBalance(String address)
-getBalance()